src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { useUID } from 'react-uid';
+import styles from './ProgressBar.module.css';
+import Text from '../Text';
+
+export type Props = {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Label associated with naming the progress bar.
+   */
+  label: string;
+  /**
+   * Max value to be represented by the progress bar.
+   */
+  max: number;
+  /**
+   * The amount of segments to be represented by the progress bar.
+   * Recommended max number of segments is 10.
+   */
+  segmentCount: number;
+  /**
+   * The value that each segment represents.
+   */
+  segmentValue: number;
+  /**
+   * An optional unit of measurement to describe the progress.
+   */
+  unit?: string;
```

Comment:
Some suggestions on prop naming is appreciated

Reply:
Yea I considered `unitOfMeasurement` but if a % sign or decimal is being used, wouldn't be applicable

Reply:
Discussed sync, will be replaced with the `caption` prop which will override pre-generated caption